types/stripe/index.d.ts

```diff
@@ -805,16 +819,30 @@ declare namespace Stripe {
             /**
              * ID of the invoice this charge is for if one exists. [Expandable]
              */
-            invoice: string | invoices.IInvoice;
+            invoice: string | invoices.IInvoice | null;
 
             livemode: boolean;
 
             metadata: IMetadata;
 
+            /**
+             * The Stripe account ID for which these funds are intended. Automatically
+             * set if you use the destination parameter. For details, see [Creating
+             * Separate Charges and Transfers]
+             * <https://stripe.com/docs/connect/charges-transfers#on-behalf-of>.
+             */
+            on_behalf_of?: string | null;
+
             /**
              * ID of the order this charge is for if one exists. [Expandable]
              */
-            order: string | orders.IOrder;
+            order: string | orders.IOrder | null;
+
+            /**
+             * Details about whether the payment was accepted, and why. See
+             * understanding declines for details. [Expandable]
+             */
+            outcome?: any;
```

Comment:
This outcome is actually a hash object with required and optional fields (https://stripe.com/docs/api#charge_object-outcome). This should be made into an interface and set that as the type for the `outcome` field.